x-pack/plugins/endpoint/common/types.ts

```diff
@@ -342,6 +342,26 @@ export interface EndpointEvent {
   };
 }
 
+/**
+ * Whitelist Rule
+ * https://github.com/elastic/endpoint-dev/issues/6532
+ */
+export interface WhitelistRule {
```

Comment:
@alexk307 will we just maintain one whitelist for all OS's in the first release?

Reply:
That was my understanding. Is that the case @ferullo?

Reply:
I wouldn't expect that. There is no field in the whitelist that indicates what OS an entry is valid for and sending the whitelist for all platforms to all endpoints would use a lot of extra bandwidth. It would be possible for Endpoint to ignore whitelist entries not for its platform if a platform field is added per-entry. But it would be better to maintain different lists per-platform.

Reply:
@stevewritescode ^^^ FYI, this conversation probably interests you too since the whitelist and manifests formats will be the same across cloud and user whitelisting. There's room for cloud and/or user to use a multi-platform whitelists and the other to use per-platform whitelists if that's what we settle on.

Reply:
We should try to deliver the whitelists artifacts per platform if possible, as there is next to zero overlap on whitelist entries across platforms.  But the most important thing is that the artifacts delivery structure from the upstream cloud artifacts and the user generated artifacts are in alignment.

Reply:
@alexk307 will we be able to tell which rules go to which OS's from the API response?  Wasn't sure if OS was represented here.

Reply:
Currently there's no separation between each rule. If this is a feature we want to prioritize, I can start creating multiple lists per operating system. I assume we can just get the OS from the generated alert that the whitelist rule was based on.

Reply:
Thanks for the heads up. We don't have any work planned right now to re-format the phase 1 malwarescore allowlist/blocklist to align with this format. We probably should, right @donaherc @bfilar?

Reply:
That is correct, though if @alexk307 has been talking to the artifact devs for these allowlists and they have changes in mind, that's fine. It'd be good to have some way to share these formats/structs between the Kibana code and whatever CI/validation tooling we have up in our publication logic so that we can make sure to ensure consistency between both going forward.

Reply:
Reading up on this issue above the struct, it does appear that there was a big involved design discussion around it. Similar to @ferullo's  stance of ensuring that departures between formats are indicated in the manifests, I think that this structure would be one of those changes. I'm completely impartial to having this be a different type of "allowlist thing" under a different name... we are using `endpointpe-whitelist` for existing allowlist, so this can be `endpoint-allowlist-p1` or something. Would love to get a programmatic interface spec so we can integrate to validation tests when they pass through publication logic.

Reply:
[This](https://github.com/elastic/endpoint-dev/issues/6532) issue's format should be the only format of allowlists/whitelists and blocklists/blacklists published for Endpoint, both for user and global lists. We need code somewhere to convert the legacy format into the new format, IMHO that should be done when the artifacts are generated, prior to being published. The new allow/block list format allows for more flexible white/black listing without Endpoint updates needed for all new whitelist targets. It would be a shame to do the translation on the Endpoint and forego that flexibility for global lists.